test/enum_if.cpp

```diff
@@ -42,7 +42,7 @@ int main()
 	io_service ios;
 	error_code ec;
 
-	address def_gw = get_default_gateway(ios, ec);
+	address def_gw = get_default_gateway(ios, "", false, ec);
```

Comment:
in order to manually test this function, do you think it would make sense to add a column (or maybe two, one for v4 and one for v6) to the enumeration of interfaces below?